src/colvar.cpp

```diff
@@ -2099,6 +2099,15 @@ bool colvar::periodic_boundaries() const
 cvm::real colvar::dist2(colvarvalue const &x1,
                          colvarvalue const &x2) const
 {
+  if ( is_enabled(f_cv_scripted) || is_enabled(f_cv_custom_function) ) {
+    if (is_enabled(f_cv_periodic) && is_enabled(f_cv_scalar)) {
+      cvm::real diff = x1.real_value - x2.real_value;
+      const double period_lower_boundary = (2.0 * wrap_center - period) / 2.0;
```

Comment:
Can you use `cvm::real` instead of `double`? This dates back to many years ago when I had a working code with Gromacs (pre-domain decomposition). Now that we have a better maintained Gromacs interface, it would be nice to test behavior when changing the underlying FP type.